### ISSUE TITLE
Update loop with git writeback in multi-image application

### ISSUE THREAD

**Issue by r0bobo** [argoproj-labs/argocd-image-updater#170]
**Describe the bug**
The image updater seems to get stuck in an infinite loop when there are multiple images to update in one Application (using Kustomize) and one of the images is currently the latest, but the other is not. It commits the latest version for one image omitting the other image fully in `.argocd-source-myapp.yaml`. Then it commits the latest version for the other image, omitting the first image, and it keeps going like that.

**To Reproduce**
# Application
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  annotations:
    argocd-image-updater.argoproj.io/image-list: coredns/coredns:~1, quay.io/coreos/etcd:~3.4
    argocd-image-updater.argoproj.io/write-back-method: git
...
```
# Kustomize
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - deploy.yaml

images:
  - name: coredns/coredns
    newTag: 1.8.0
  - name: quay.io/coreos/etcd
    newTag: v3.4.0
```

# .argocd-source-myapp.yaml
```yaml
kustomize:
  images:
  - coredns/coredns:1.8.3 # (latest version at the time)
```

**Expected behavior**
I expect the latest version of both images to be present in `.argocd-source-myapp.yaml` at all times.

**Additional context**
Git commit history:
```diff
commit 6626f849a90c1736880265b05266e564e32ab323
Author: argocd-image-updater <noreply@argoproj.io>
Date:   Tue Mar 23 23:02:33 2021 +0000

    Update to new image versions

diff --git a/.argocd-source-argocd-image-updater-dummy.yaml b/.argocd-source-argocd-image-updater-dummy.yaml
index e49c21a..a0fc1b1 100644
--- a/.argocd-source-argocd-image-updater-dummy.yaml
+++ b/.argocd-source-argocd-image-updater-dummy.yaml
@@ -1,3 +1,3 @@
 kustomize:
   images:
-  - coredns/coredns:1.8.3
+  - quay.io/coreos/etcd:v3.4.15

commit e23933ca78a26e03e037e74ad36bc2854f1c335d
Author: argocd-image-updater <noreply@argoproj.io>
Date:   Tue Mar 23 22:56:19 2021 +0000

    Update to new image versions

diff --git a/.argocd-source-argocd-image-updater-dummy.yaml b/.argocd-source-argocd-image-updater-dummy.yaml
index dfae843..e49c21a 100644
--- a/.argocd-source-argocd-image-updater-dummy.yaml
+++ b/.argocd-source-argocd-image-updater-dummy.yaml
@@ -1,4 +1,3 @@
 kustomize:
   images:
-  - coredns/coredns:1.8.0
-  - quay.io/coreos/etcd:v3.4.15
+  - coredns/coredns:1.8.3
```

**Version**
`argocd-image-updater:v0.9.0`

**Logs**
```
time="2021-03-23T22:58:24Z" level=info msg="Processing results: applications=1 images_considered=2 images_skipped=0 images_updated=1 errors=0"
time="2021-03-23T23:00:24Z" level=info msg="Starting image update cycle, considering 1 annotated application(s) for update"
time="2021-03-23T23:00:28Z" level=info msg="Setting new image to coredns/coredns:1.8.3" alias= application=myapp image_name=coredns/coredns image_tag=1.8.0 registry=
time="2021-03-23T23:00:28Z" level=info msg="Successfully updated image 'coredns/coredns:1.8.0' to 'coredns/coredns:1.8.3', but pending spec update (dry run=false)" alias= application=myapp image_name=coredns/coredns image_tag=1.8.0 registry=
time="2021-03-23T23:00:29Z" level=info msg="Committing 1 parameter update(s) for application myapp" application=myapp
time="2021-03-23T23:00:29Z" level=info msg="Starting configmap/secret informers"
time="2021-03-23T23:00:29Z" level=info msg="Configmap/secret informer synced"
time="2021-03-23T23:00:29Z" level=info msg="configmap informer cancelled"
time="2021-03-23T23:00:29Z" level=info msg="secrets informer cancelled"
time="2021-03-23T23:00:29Z" level=info msg="Initializing https://git.todevski.com/dean/myapp.git to /tmp/git-myapp612143736"
time="2021-03-23T23:00:29Z" level=info msg="rm -rf /tmp/git-myapp612143736" dir= execID=EFrgR
time="2021-03-23T23:00:29Z" level=info msg="git fetch origin --tags --force" dir=/tmp/git-myapp612143736 execID=odQZx
time="2021-03-23T23:00:29Z" level=info msg="git config user.name argocd-image-updater" dir=/tmp/git-myapp612143736 execID=2mbsR
time="2021-03-23T23:00:29Z" level=info msg="git config user.email noreply@argoproj.io" dir=/tmp/git-myapp612143736 execID=62PSv
time="2021-03-23T23:00:29Z" level=info msg="git checkout --force main" dir=/tmp/git-myapp612143736 execID=eO1Ax
time="2021-03-23T23:00:29Z" level=info msg="git clean -fdx" dir=/tmp/git-myapp612143736 execID=9hKPP
time="2021-03-23T23:00:29Z" level=info msg="Successfully updated the live application spec" application=myapp
time="2021-03-23T23:00:29Z" level=info msg="Processing results: applications=1 images_considered=2 images_skipped=0 images_updated=1 errors=0"
time="2021-03-23T23:02:29Z" level=info msg="Starting image update cycle, considering 1 annotated application(s) for update"
time="2021-03-23T23:02:33Z" level=info msg="Setting new image to quay.io/coreos/etcd:v3.4.15" alias= application=myapp image_name=coreos/etcd image_tag=v3.4.0 registry=quay.io
time="2021-03-23T23:02:33Z" level=info msg="Successfully updated image 'quay.io/coreos/etcd:v3.4.0' to 'quay.io/coreos/etcd:v3.4.15', but pending spec update (dry run=false)" alias= application=myapp image_name=coreos/etcd image_tag=v3.4.0 registry=quay.io
time="2021-03-23T23:02:33Z" level=info msg="Committing 1 parameter update(s) for application myapp" application=myapp
time="2021-03-23T23:02:33Z" level=info msg="Starting configmap/secret informers"
time="2021-03-23T23:02:33Z" level=info msg="Configmap/secret informer synced"
time="2021-03-23T23:02:33Z" level=info msg="Initializing https://git.todevski.com/dean/myapp.git to /tmp/git-myapp114701175"
time="2021-03-23T23:02:33Z" level=info msg="secrets informer cancelled"
time="2021-03-23T23:02:33Z" level=info msg="configmap informer cancelled"
time="2021-03-23T23:02:33Z" level=info msg="rm -rf /tmp/git-myapp114701175" dir= execID=1IYjj
time="2021-03-23T23:02:33Z" level=info msg="git fetch origin --tags --force" dir=/tmp/git-myapp114701175 execID=4YK5l
time="2021-03-23T23:02:33Z" level=info msg="git config user.name argocd-image-updater" dir=/tmp/git-myapp114701175 execID=Dj1on
time="2021-03-23T23:02:33Z" level=info msg="git config user.email noreply@argoproj.io" dir=/tmp/git-myapp114701175 execID=h6Dzp
time="2021-03-23T23:02:33Z" level=info msg="git checkout --force main" dir=/tmp/git-myapp114701175 execID=U9vjc
time="2021-03-23T23:02:33Z" level=info msg="git clean -fdx" dir=/tmp/git-myapp114701175 execID=rD3hf
time="2021-03-23T23:02:33Z" level=info msg="git commit -a -m Update to new image versions" dir=/tmp/git-myapp114701175 execID=b2ZSA
time="2021-03-23T23:02:34Z" level=info msg="git push origin main" dir=/tmp/git-myapp114701175 execID=04Ose
time="2021-03-23T23:02:34Z" level=info msg="Successfully updated the live application spec" application=myapp
time="2021-03-23T23:02:34Z" level=info msg="Processing results: applications=1 images_considered=2 images_skipped=0 images_updated=1 errors=0"
```

**Comment by jannfis** [comment]
Hey @r0bobo, thanks for the detailed bug report! I could successfully reproduce it.

**Comment by jannfis** [comment]
@r0bobo I just released [v0.9.1](https://github.com/argoproj-labs/argocd-image-updater/releases/tag/v0.9.1) which fixes this problem. Feel free to re-open this issue if it does not.

**Comment by r0bobo** [comment]
Awesome!
